typescript/sdk/src/token/deploy.ts

```diff
@@ -219,7 +208,7 @@ export class HypERC20Deployer extends GasRouterDeployer<
       throw new Error('Invalid ERC20 token router config');
     }
     await this.configureClient(chain, router, config);
-    return { router };
+    return { [getTokenType(config.type, TokenStandard.ERC20)]: router } as any;
```

Comment:
Can we use a generic type in this deployContracts method to avoid the multiple new casts we have in this file?

Reply:
++

Reply:
I've tried TokenRouter but it's exactly work without casting